otel-integration/k8s-helm/values.yaml

```diff
@@ -0,0 +1,470 @@
+      #   gcp:
+      #     resource_attributes:
+      #       cloud.region:
+      #         enabled: true
+      #       cloud.availability_zone:
+      #         enabled: true
+      #   ec2:
+      #     resource_attributes:
+      #       cloud.region:
+      #         enabled: true
+      #       cloud.availability_zone:
+      #         enabled: true
+      metricstransform:
+        transforms:
+          include: .*
+          match_type: regexp
+          action: update
+          operations:
+            - action: add_label
+              new_label: k8s.cluster.name
+              new_value: "{{ .Values.global.clusterName }}"
+            - action: add_label
+              new_label: cx.otel_integration.name
+              new_value: "{{ .Chart.Name }}"
+            - action: add_label
+              new_label: cx.otel_integration.version
+              new_value: "{{ .Chart.Version }}"
+      k8sattributes:
+        filter:
+          node_from_env_var: KUBE_NODE_NAME     
+        extract:
+          metadata:
+            - "k8s.namespace.name"
+            - "k8s.deployment.name"
+            - "k8s.statefulset.name"
+            - "k8s.daemonset.name"
+            - "k8s.cronjob.name"
+            - "k8s.job.name"
+            - "k8s.pod.name"
+            - "k8s.node.name"
+      spanmetrics:
+        metrics_exporter: coralogix
+        dimensions:
+          - name: "k8s.deployment.name"
+          - name: "k8s.statefulset.name"
+          - name: "k8s.daemonset.name"
+          - name: "k8s.cronjob.name"
+          - name: "k8s.job.name"
+          - name: "k8s.container.name"
+          - name: "k8s.node.name"
+          - name: "k8s.namespace.name" 
+      memory_limiter: null # Will get the k8s resource limits
+
+    exporters:
+      coralogix:
+        timeout: "30s"
+        private_key: "${CORALOGIX_PRIVATE_KEY}"
+        domain: "{{ .Values.global.domain }}"
+        application_name: "{{ .Values.global.defaultApplicationName }}"
+        subsystem_name: "{{ .Values.global.defaultSubsystemName }}"
+        application_name_attributes:
+          - "k8s.namespace.name"
+          - "service.namespace"
+        subsystem_name_attributes:
+          - "k8s.deployment.name"
+          - "k8s.statefulset.name"
+          - "k8s.daemonset.name"
+          - "k8s.cronjob.name"
+          - "k8s.job.name"
+          - "k8s.container.name"
+          - "k8s.node.name"
+          - "service.name"
+
+    service:
+      telemetry:
+        logs:
+          level: "{{ .Values.global.logLevel }}"
+          encoding: json
+        metrics:
+          address: ${MY_POD_IP}:8888
+      extensions:
+      - zpages
+      - pprof
+      - health_check
+      - memory_ballast
+      pipelines:
+        metrics:
+          exporters:
+            - coralogix
+          processors:
+            - k8sattributes
+            - resourcedetection/env
+            # - resourcedetection/region
+            - metricstransform
```

Comment:
On the monitoring shippers we are ensuring that we will not collect more than is needed for the K8S Dashboard work.

https://github.com/coralogix/monitoring-shippers/blob/main/charts/values.yaml#L146

Should we keep this behavior here?

Reply:
So it might be a bit confusing here, but this is now:
- For cluster receiver metrics, we don't filter out, since if `clusterMetrics` preset is enabled, we send metrics from that receiver (based on the preset config)
- For the extra metrics (KSM and API / node metrics), these are filtered in the chart as part of the `kubernetesExtraMetrics` preset (see [here](https://github.com/coralogix/opentelemetry-helm-charts/blob/main/charts/opentelemetry-collector/templates/_config.tpl#L195) and [here](https://github.com/coralogix/opentelemetry-helm-charts/blob/main/charts/opentelemetry-collector/templates/_config.tpl#L378))

We perhaps could do some more complex checking (i.e. when K8s dashboard extension is enabled, but cluster metrics disabled, we would only collect necessary metrics for dashboard), but I did not find a good way of doing this 😒. So for now it is kind of assumed that we collect all cluster receiver metrics, if this preset is enabled, regardless of the k8s dashboard parameter. I hope that works.

Reply:
I guess this is Ok